app/api/api/views.py

```diff
@@ -390,8 +391,33 @@ class ScanReportListViewSetV2(ScanReportListViewSet):
     def get_serializer_class(self):
         if self.request.method in ["GET", "POST"]:
             return ScanReportViewSerializerV2
+        if self.request.method in ["DELETE"]:
+            return ScanReportTableEditSerializer
         return super().get_serializer_class()
 
+    def destroy(self, request, pk):
+        instance = self.get_object()
+        required_permissions = ["CanView", "CanEdit", "CanAdmin"]
+        permissions = get_user_permissions_on_scan_report(request, pk)
```

Comment:
I hadn't realised, but there's no need for you to handle permissions. This is done on the base class, line 245. So you can delete any permission related code here :)